os/drivers/serial/serial.c

```diff
@@ -125,6 +139,86 @@ static const struct file_operations g_serialops = {
  * Private Functions
  ************************************************************************************/
 
+#ifdef CONFIG_DYNAMIC_LOGS_STORING
+void read_log_net(void)
+{	size_t free_size;
+
+	if (!sq_empty(&log_net_chunks)) { /* check if there are any logs saved */
+		struct mm_heap_s *myheap = mm_get_heap(sq_peek(&log_net_chunks)); /* get heap details using the head of the chunks linked list */
+		if (myheap == NULL) {
+			lldbg("mm_get_heap failed\n");
+			return;
+		}
+		free_size = myheap->mm_heapsize - myheap->total_alloc_size; /* get the free size */
+	} else { /* we have not logs stored, so return */
+		return;
+	}
+
+	lldbg("Read output ---> starts here <--- log size : %d, free heap : %d\n", log_net_size, free_size);
```

Comment:
Cosmetic change suggestion
Can we have a line of asterisks to demarcate the start and end point of the logs clearly?
Like: - 
```
***********************************************************
Logs start
.
.
.
Logs End
***********************************************************
```